tools/block_cache_analyzer/block_cache_trace_analyzer.cc

```diff
@@ -1209,11 +1211,14 @@ void BlockCacheTraceAnalyzer::WriteBlockReuseTimeline(
   TraverseBlocks(block_callback);
 
   // A cell is the number of blocks accessed in a reuse window.
-  uint64_t reuse_table[reuse_vector_size][reuse_vector_size];
+  const std::unique_ptr<uint64_t[]> reuse_table(
```

Comment:
Curious, MSVC forbids two-dimensional arrays now? Can @hliu18 or @burtonli verify this behavior?

Reply:
MSVC supports two-dimensional arrays, but does not support variable length arrays, i.e. dimension sizes need to be known at compile time. Variable length array is not part of recent versions of C++ standard.

Reply:
Will the following compile and run on MSVC?
```
#include <cstdio>

int main(int argc, char** argv) {
  const size_t sz = argc;
  int test_arr[sz];
  fprintf(stdout, "size is %d\n", sizeof(test_arr));
  return 0;
}
```
If not, this change is necessary.
It probably won't, according to your previous comment about compile-time.

Reply:
It does not compile: "error C2131: expression did not evaluate to a constant"
The compiler complains about [sz].

Reply:
Thanks @hliu18 for verifying this.

Reply:
I see this PR was imported but not checked in due to conflicts. Can someone help move this PR forward? I was going to fix a couple of build break on MSVC and noticed this.